avahi-core/server.c

```diff
@@ -1305,17 +1309,28 @@ int avahi_server_set_host_name(AvahiServer *s, const char *host_name) {
     else
         hn = avahi_normalize_name_strdup(host_name);
 
-    hn[strcspn(hn, ".")] = 0;
+    h = hn;
```

Comment:
I think `hn` can be `NULL` here. It's weird that coverity has never complained about it.